cmd/btcatomicswap/main.go

```diff
@@ -452,6 +452,60 @@ func fundRawTransaction(c *rpc.Client, tx *wire.MsgTx, feePerKb btcutil.Amount)
 	return fundedTx, feeAmount, nil
 }
 
+// signRawTransaction calls the signRawTransaction JSON-RPC method.  It is
+// implemented manually as client support is currently outdated from the
+// btcd/rpcclient package.
+func signRawTransaction(c *rpc.Client, tx *wire.MsgTx) (fundedTx *wire.MsgTx, complete bool, err error) {
+	var buf bytes.Buffer
+	buf.Grow(tx.SerializeSize())
+	tx.Serialize(&buf)
+	param, err := json.Marshal(hex.EncodeToString(buf.Bytes()))
+	if err != nil {
+		return nil, false, err
+	}
+	rawResp, err := c.RawRequest("signrawtransactionwithwallet", []json.RawMessage{param})
+	if err != nil {
+		return nil, false, err
+	}
+	var resp struct {
+		Hex      string `json:"hex"`
+		Complete bool   `json:"complete"`
+	}
+	err = json.Unmarshal(rawResp, &resp)
+	if err != nil {
+		return nil, false, err
+	}
+	fundedTxBytes, err := hex.DecodeString(resp.Hex)
+	if err != nil {
+		return nil, false, err
+	}
+	fundedTx = &wire.MsgTx{}
+	err = fundedTx.Deserialize(bytes.NewReader(fundedTxBytes))
+	if err != nil {
+		return nil, false, err
+	}
+	return fundedTx, resp.Complete, nil
+}
+
+// sendRawTransaction calls the signRawTransaction JSON-RPC method.  It is
+// implemented manually as client support is currently outdated from the
+// btcd/rpcclient package.
+func sendRawTransaction(c *rpc.Client, tx *wire.MsgTx) (string, error) {
+	var buf bytes.Buffer
+	buf.Grow(tx.SerializeSize())
+	tx.Serialize(&buf)
+
+	param, err := json.Marshal(hex.EncodeToString(buf.Bytes()))
+	if err != nil {
+		return "", err
+	}
+	hex, err := c.RawRequest("sendrawtransaction", []json.RawMessage{param})
+	if err != nil {
+		return "", err
+	}
+	return string(hex), nil
```

Comment:
this returns a string with json quotes still present.  we should return a chainhash.Hash or a byte slice instead (chainhash.Hash is preferred since it's also a fmt.Stringer so it can be printed properly, and converting to it will perform a length sanity check).

Reply:
fixed